pkg/cmd/cli/kubeflowmodelregistry/kfmr.go

```diff
@@ -452,9 +453,20 @@ func (m *ModelPopulator) GetTechDocs() *string {
 	if techdocsUrl == nil && strings.Contains(m.GetName(), brdgtypes.Granite318bLabName) {
 		granite31TechDocs := brdgtypes.Granite318bLabTechDocs
 		return &granite31TechDocs
-	} else {
-		return techdocsUrl
+	} else if techdocsUrl != nil {
+		u, err := url.Parse(*techdocsUrl)
+		switch {
+		case err != nil:
+			fallthrough
+		case u == nil:
+			fallthrough
+		case u != nil && (u.Scheme != "http" && u.Scheme != "https"):
+			klog.Errorf("returning default techdoc URL since there is either an error or bad scheme for techdoc url %v: err %v, url %v", techdocsUrl, err, u)
```

Comment:
IMO, we should default to no techdocs here rather than Granite if the URL parsing fails, especially if it's a non granite model

Reply:
agreed / switched returning nil instead on error etc.